src/data/list/basic.lean

```diff
@@ -2945,6 +2945,25 @@ theorem map_filter_map_of_inv (f : α → option β) (g : β → α)
   map g (filter_map f l) = l :=
 by simp only [map_filter_map, H, filter_map_some]
 
+theorem map_filter_map_some_eq_filter_map_is_some (f : α → option β) (l : list α) :
```

Comment:
Can you move this up to right below `filter_map_some`? These feel like very related lemmas.